program-runtime/src/invoke_context.rs

```diff
@@ -197,6 +197,7 @@ pub struct InvokeContext<'a> {
     pub timings: ExecuteDetailsTimings,
     pub blockhash: Hash,
     pub lamports_per_signature: u64,
+    pub processed_inner_instructions: Vec<(usize, Instruction)>,
```

Comment:
This should go in `TransactionContext`, because it is not runtime internal and should be exposed to ABIv2 programs as well. I am preparing a PR that moves the `InstructionRecorder` there, on which you could rebase this PR.

Reply:
I'm going to merge this today, feel free to move it after